agr_literature_service/api/tests/test_017_reference_ontology.py

```diff
@@ -0,0 +1,163 @@
+import pytest
+from fastapi import HTTPException
+from pydantic import ValidationError
+from sqlalchemy import MetaData, create_engine
+from sqlalchemy.orm import sessionmaker
+
+from agr_literature_service.api.crud.reference_ontology_crud import (
+    create, destroy, patch, show, show_changesets)
+from agr_literature_service.api.database.config import SQLALCHEMY_DATABASE_URL
+from agr_literature_service.api.database.base import Base
+from agr_literature_service.api.models import (ReferenceOntologyModel,
+                                               ReferenceModel)
+from agr_literature_service.api.schemas import ReferenceOntologySchemaCreate
+from agr_literature_service.api.crud.mod_crud import create as mod_create
+from agr_literature_service.api.crud.user_crud import create as user_create
+
+metadata = MetaData()
+
+engine = create_engine(SQLALCHEMY_DATABASE_URL, connect_args={"options": "-c timezone=utc"})
+SessionLocal = sessionmaker(bind=engine, autoflush=True)
+db = SessionLocal()
+
+# Add tables/schema if not already there.
+Base.metadata.create_all(engine)
+
+# Exit if this is not a test database, Exit.
+if "literature-test" not in SQLALCHEMY_DATABASE_URL:
+    exit(-1)
+
+fb_mod = None
+
+
+def test_initialise():
+    global fb_mod
+    # add User "017 Bob"
+    user_create(db, "017_Bob")
+
+    # add mods
+    data = {
+        "abbreviation": '017_FB',
+        "short_name": "017_FB",
+        "full_name": "017_ont_1"
+    }
+    fb_mod = mod_create(db, data)
+
+    data = {
+        "abbreviation": '017_RGD',
+        "short_name": "017_Rat",
+        "full_name": "017_ont_2"
+    }
+    mod_create(db, data)
+
+
+def test_get_bad_ref_ont():
+    with pytest.raises(HTTPException):
+        show(db, 99999)
+
+
+def test_bad_missing_args():
+    xml = {'reference_curie': "AGR:AGR-Reference-0000000001",
+           'ontology_id': "ont1",
+           'created_by': "017_Bob"}
+    with pytest.raises(ValidationError):  # mod_abbr missing
```

Comment:
I'm pretty sure they want to be able to not put in a MOD, and have null mean all mods.  Does this mean that ReferenceOntologySchemaCreate having      mod_abbreviation: str  instead of      mod_abbreviation: Optional[str], require that create a referenceOntology entry must pass a mod_abbreviation: null, or that it cannot allow a mod_abbreviation: null ?  (If it requires passing an explicit null, that's fine, if not, it's probably not right)

Reply:
Good spot, I need to test this and see what happens.

Reply:
Tested, you need to pass "mod_abbreviation": "", if you to not specify the mod (i.e. all mods)